### DISCUSSION TITLE
Moes ZK-CH-2U (ZK-XX-2U) as Model HGZB-01, Vendor Nue / 3A recognized

### DISCUSSION THREAD

**Discussion by cflurin** [Koenkk/zigbee2mqtt#5817] [Devices]
Hi,
I've added a new device Moes ZK-CH-2U (ZK-XX-2U) but the device has been recognized as Model HGZB-01, Vendor Nue / 3A.
The device works but the pic, the Model and the Vendor are wrong.

zigbee2mqtt version: 1.17.0
coordinator: CC2531

Any tips how to modify my database.db

```
{
	"id": 11,
	"type": "Router",
	"ieeeAddr": "0x0xxxxxxxxxxxxxxx",
	"nwkAddr": 35242,
	"manufId": 4478,
	"manufName": "3A Smart Home DE",
	"powerSource": "Mains (single phase)",
	"modelId": "LXN59-1S7LX1.0",
	"epList": [1],
	"endpoints": {
		"1": {
			"profId": 260,
			"epId": 1,
			"devId": 256,
			"inClusterList": [0, 3, 4, 5, 6],
			"outClusterList": [],
			"clusters": {
				"genBasic": {
					"attributes": {
						"modelId": "LXN59-1S7LX1.0",
						"manufacturerName": "3A Smart Home DE",
						"powerSource": 0,
						"zclVersion": 3,
						"appVersion": 1,
						"stackVersion": 0,
						"hwVersion": 0,
						"dateCode": "20200519"
					}
				},
				"genOnOff": {
					"attributes": {
						"onOff": 1
					}
				}
			},
			"binds": [{
				"cluster": 6,
				"type": "endpoint",
				"deviceIeeeAddress": "0x0xxxxxxxxxxxxxxx",
				"endpointID": 1
			}],
			"configuredReportings": [{
				"cluster": 6,
				"attrId": 0,
				"minRepIntval": 0,
				"maxRepIntval": 3600,
				"repChange": 0
			}],
			"meta": {}
		}
	},
	"appVersion": 1,
	"stackVersion": 0,
	"hwVersion": 0,
	"dateCode": "20200519",
	"zclVersion": 3,
	"interviewCompleted": true,
	"meta": {
		"configured": 2
	},
	"lastSeen": 1611312292649
}
```

**Comment by infeeeee** [comment]
I also have this problem, but it doesn't work as it should, it's not only a name and picture problem. I guess the usbs and the plug could be switched separately, but for my case I can just see one switch, so the 230V and the usbs are switched together.

Did you found a solution for this? Should we open an issue about this?

**Reply by cflurin** [reply]
Hi,
No, I didn't find a solution yet but for me it's temporarly okey. I use the moes Plug mostly as router and as charger for my iPhone/iPad.

I think, you should open an issue if you need to separately control the usbs.

**Reply by infeeeee** [reply]
Thanks, I will open an issue soon, but first I will try to solve it myself.

Other info for others later looking at this thread in the future:
Maybe this issue is related to this, as it's the same model:
https://github.com/Koenkk/zigbee2mqtt/issues/4503

Here is the log when I join the zigbee network:

```
Zigbee2MQTT:info  2021-02-03 03:01:22: Device '0x0xxxxxxxxxxxxxxx' joined                                                         
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x0xxxxxxxxxxxxxxx","ieee_address":"0x0xxxxxxxxxxxxxxx"},"type":"device_joined"}'                                                                                                 
Zigbee2MQTT:info  2021-02-03 03:01:22: Starting interview of '0x0xxxxxxxxxxxxxxx'                                                 
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x0xxxxxxxxxxxxxxx","ieee_address":"0x0xxxxxxxxxxxxxxx","status":"started"},"type":"device_interview"}'                                                                           
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0x0xxxxxxxxxxxxxxx"},"type":"device_connected"}'                                                                                                                                
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0x0xxxxxxxxxxxxxxx"},"type":"pairing"}'                                                                                                              
Zigbee2MQTT:debug 2021-02-03 03:01:22: Device '0x0xxxxxxxxxxxxxxx' announced itself                                              
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x0xxxxxxxxxxxxxxx","ieee_address":"0x0xxxxxxxxxxxxxxx"},"type":"device_announce"}'                                                                                              
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x0xxxxxxxxxxxxxxx"},"type":"device_announced"}'                                                                                                              
Zigbee2MQTT:debug 2021-02-03 03:01:22: Device '0x0xxxxxxxxxxxxxxx' announced itself                                              
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x0xxxxxxxxxxxxxxx","ieee_address":"0x0xxxxxxxxxxxxxxx"},"type":"device_announce"}'                                                                                              
Zigbee2MQTT:info  2021-02-03 03:01:22: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x0xxxxxxxxxxxxxxx"},"type":"device_announced"}'                                                                                                              
Zigbee2MQTT:debug 2021-02-03 03:01:23: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"modelId":"LXN59-1S7LX1.0"}' from endpoint 1 with groupID 0                                                                                                      
Zigbee2MQTT:debug 2021-02-03 03:01:23: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"3A Smart Home DE"}' from endpoint 1 with groupID 0                
Zigbee2MQTT:debug 2021-02-03 03:01:23: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"powerSource":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-02-03 03:01:23: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"zclVersion":3}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-02-03 03:01:24: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"appVersion":1}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-02-03 03:01:24: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"stackVersion":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-02-03 03:01:24: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"hwVersion":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-02-03 03:01:24: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{"dateCode":"20200519"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-02-03 03:01:24: Received Zigbee message from '0x0xxxxxxxxxxxxxxx', type 'readResponse', cluster 'genBasic', data '{}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-02-03 03:01:24: Successfully interviewed '0x0xxxxxxxxxxxxxxx', device has successfully been paired
Zigbee2MQTT:info  2021-02-03 03:01:24: Device '0x0xxxxxxxxxxxxxxx' is supported, identified as: Nue / 3A Smart Zigbee 3.0 light controller (HGZB-01)
```

**Reply by cflurin** [reply]
There are several `Dual USB Plugs`:

https://zigbee.blakadder.com/Moes_ZK-XX-2U.html

I have the CH Type, which one are you using?

**Reply by infeeeee** [reply]
I have the EU version, but I guess it shouldn't matter

I bought it from Aliexpress from 'MoesHouse Official Store': https://www.aliexpress.com/item/4000174314753.html

I modified devices.js to fake it as the moes device, but it didn't work. I even tried to fake as a [TS0108](https://www.zigbee2mqtt.io/devices/TS0108.html), but it didn't work either. I can't find any other similar supported devices.

I'm starting to think it's a counterfeit device or a crippled new revision. It's strange that nor manufacturer's name nor the device name is not written on the device or on the box.
![IMG_20210203_142002_1](https://user-images.githubusercontent.com/1271737/106754665-87a17900-662d-11eb-8a41-e9e9440ef7ad.jpg)

![IMG_20210203_141942_1](https://user-images.githubusercontent.com/1271737/106754647-81ab9800-662d-11eb-90ec-37e779c9d84d.jpg)

![IMG_20210203_142020_1](https://user-images.githubusercontent.com/1271737/106754688-8e2ff080-662d-11eb-9d05-fbf9c86c866c.jpg)

I will disassemble to check what's inside: if it has only one relay than I would gave up, as it's physically not possible to switch them separately

**Reply by schmoogol** [reply]
Did you make any more progress with this? I've bought the **ZK-AU-2U** from the the MoesHouse AliExpress store and it has the same issue, reporting as model '**LXN59-1S7LX1.0**' instead of '**TS0112**'. I suspect that if I took it apart it would look the same as yours on the inside with the updated PCB date.

Did you find any alternative plugs that do offer discrete control over USB and power or somewhere that sells the old (or non-counterfeit) TS0112 version?

**Reply by infeeeee** [reply]
Unfortunately no, I just use this as a regular plug, I don't even use the usb port...

I just found this thread, it seems new versions can't control the usbs separately: https://github.com/Koenkk/zigbee2mqtt/issues/6354

**Comment by infeeeee** [comment]
(I start a new comment thread because Github hides long comment threads, and I think this is important information)

I disassembled this device, disassembly was quite easy: There is only 1 screw under the vignette, after unscrewing this there are only plastic pins around the body, they can be opened with a flat plastic card. The PCB can be unscrewed with 2 other screws.

It seems like it's a new revision, this is written on the PCB: 
```
XN-WGCZ+2USB T1.3
2020.06.04
```

It seems there is only one relay inside, so I think it's not possible to switch them separately. But I'm not pro with these electronic things, so it's possible that I oversight something.

I added some annotation to the photos, so maybe it can help someone with more knowledge about electronics:

### Front
🟧: Zhejiang Fanhar FH17-1A2TLE-DC5V Relay
🟥: Vertical daughter board, I think this is the Zigbee radio, shielded, antenna is clearly visible, text on a vignette: `1S710-2011`
🟩: Some inductor
🟪: Some pins are unpopulated here. It's possible that the second relay should be here?

![PCB_front_annotated](https://user-images.githubusercontent.com/1271737/106798868-de737680-665e-11eb-9263-908c7f4ebb35.jpg)

Without annotations:

![IMG_20210203_152216_1](https://user-images.githubusercontent.com/1271737/106799008-0662da00-665f-11eb-9db2-66c2a0e70004.jpg)

Closer look:

![IMG_20210203_152242_1](https://user-images.githubusercontent.com/1271737/106799050-1084d880-665f-11eb-97a8-a136e41bee66.jpg)

The daughter board:

![IMG_20210203_152300_1](https://user-images.githubusercontent.com/1271737/106799142-2f836a80-665f-11eb-9b36-190acce7038f.jpg)


### Back
🟧: AMS1117 1A LOW DROPOUT VOLTAGE REGULATOR, fixed 5V, so it's for the USB
🟥: MB10F SMD smd bridge rectifier
🟩: OB2500PCP High Precision CC/CV Primary-Side Power Switch
🟣: Connectors to 230V on the other side
🟪: Relay on other side
🟨: Connectors to USBs

![PCB_back_annotated](https://user-images.githubusercontent.com/1271737/106797342-c00c7b80-665c-11eb-8837-fc58908bc7b0.jpg)

Without annotation:

![PCB_back](https://user-images.githubusercontent.com/1271737/106797368-ca2e7a00-665c-11eb-952f-1cc41f6c5644.jpg)

**Comment by cflurin** [comment]
On my Dual Plug label the name is `CH Zigbee 3.0 Smart Socket` so I suppose the model is `ZK-CH-2U`
 
![IMG_5742](https://user-images.githubusercontent.com/5056710/106858230-a18c9b80-66c1-11eb-97fa-a18174f682d1.jpg)

**Comment by infeeeee** [comment]
@FaBRiK74 I saw on your issue that you have a better working device. Can you disassemble and create some photos how it looks like inside? I want to compare this revision to yours, as it's possible that it's not even possible to switch the usbs on this revision separately. Thank you. There is only one screw under the sticker. If you don't want to disassemble it, I understand.
